proud-puffins/djangoProject/earlydating/puffin_functions.py

```diff
@@ -0,0 +1,39 @@
+from PIL import Image
+from django.core.files import File
+from django.core.exceptions import ValidationError
+from io import BytesIO
+import os
+
+
+def imageTrans(image):
+    try:
+        _, file_extension = os.path.splitext(image.name)
+        file_extension = file_extension.split(".")[-1]
+
+        img = Image.open(image)
+
+        imgSmall = img.resize(
+            (img.size[0] // 3, img.size[1] // 3), resample=Image.BILINEAR
+        )
+
+        result = imgSmall.resize(img.size, Image.NEAREST)
+
+        pixel_io = BytesIO()
+
+        result.save(pixel_io, "JPEG" if file_extension == "jpg" else file_extension)
+
+        pixel = File(pixel_io, name=image.name)
+
+        return pixel
+
+    except Exception:
+        return image
+
```

Comment:
While it may be that the idea here is to transform the image in some way if you can, but that it doesn't matter if you can't, you will still want to figure out which exception(s) you'd expect to get if you can't do what you wanted and only handle those.

In this section, any exception that is raised in this function or in any of the other functions it calls gets handled, even if that exception was the result of something you wrote being incorrect. You wouldn't want to recover from those exceptions because you want to know what mistakes exist in your code.